pkg/cloud-controller-manager/loadBalancer.go

```diff
@@ -69,15 +70,14 @@ func (l *LoadBalancerManager) GetLoadBalancerName(ctx context.Context, clusterNa
 
 	digest := crc32.ChecksumIEEE([]byte(name + string(service.UID)))
 	suffix := fmt.Sprintf("%08x", digest) // print in 8 width and pad with 0's
-	name += suffix
 
 	// The name of a Service object must be a valid [RFC 1035 label name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names)
 	// The name contains no more than 63 characters
-	if len(name) > maxNameLength {
-		name = name[:maxNameLength]
+	if len(name) > maxNameLength-lenOfSuffix {
+		name = name[:maxNameLength-lenOfSuffix]
```

Comment:
can u help to add a unit test for it, also the name may still contain invalid characters like `-` after slices.

Reply:
`-` is not an invalid character, but could not appear in the end.